presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java

```diff
@@ -245,7 +250,10 @@ private Page processColumnSource()
             operatorContext.recordGeneratedInput(bytesProcessed, output.getProcessedRows(), elapsedNanos);
             completedBytes = cursor.getCompletedBytes();
             readTimeNanos = cursor.getReadTimeNanos();
-            finishing = output.isNoMoreRows();
+            if (output.isNoMoreRows()) {
```

Comment:
`processColumnSource` uses `pageBuilder` as a buffer. There is no need for merging output.